extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/dataspaceconnector/sql/assetindex/SqlAssetIndex.java

```diff
@@ -229,14 +231,9 @@ private Object fromPropertyValue(String value, String type) throws ClassNotFound
     }
 
     private boolean existsById(String assetId, Connection connection) {
-        var assetCount = transactionContext.execute(() -> executeQuery(connection, this::mapRowCount, assetStatements.getCountAssetByIdClause(), assetId).iterator().next());
-
-        if (assetCount <= 0) {
-            return false;
-        } else if (assetCount > 1) {
-            throw new IllegalStateException("Expected result set size of 0 or 1 but got " + assetCount);
-        } else {
-            return true;
+        var sql = assetStatements.getCountAssetByIdClause();
+        try (var stream = SqlQueryExecutor.executeQuery(connection, false, this::mapRowCount, sql, assetId)) {
```

Comment:
I don't think this has to be a `Stream`. It executes a simple `SELECT COUNT (*)` query, which returns an int. Seems we're killing rats with a cannon.